### PR TITLE
Support tracking git references in `track bundle`

### PR DESCRIPTION
**Pull request by zregvart**
Also contains some refactoring in separate commits.

### CODE REVIEW

File: internal/tracker/tracker.go [hunk 1]
```diff
@@ -182,15 +223,30 @@ func Track(ctx context.Context, urls []string, input []byte, prune bool, freshen
...
+}
 
-	t.setExpiration()
+func trackGitReferences(_ context.Context, urls []string, t *Tracker) error {
+	effective_on := effectiveOn()
 
-	if err := t.convertToOldFormat(); err != nil {
-		return nil, err
+	for _, u := range urls {
+		repository, rest, found := strings.Cut(u, "//")
```
Comment (joejstuart) [inline internal/tracker/tracker.go:233]: Does this mean we have to specify `git+ssh` or `git+https` for task references (used by git resolver) when calling the `ec track`  command?

Comment (lcarva) [reply internal/tracker/tracker.go:233]: https://github.com/enterprise-contract/ec-policies/blob/main/example/data/trusted_tekton_tasks.yml#L3

File: cmd/track/track_bundle.go [hunk 2]
```diff
@@ -170,7 +166,10 @@ func trackBundleCmd(track trackBundleFn, pullImage pullImageFn, pushImage pushIm
 	cmd.Flags().StringVarP(&params.input, "input", "i", params.input, "existing tracking file")
 
 	cmd.Flags().StringSliceVarP(&params.bundles, "bundle", "b", params.bundles,
-		"bundle image reference to track - may be used multiple times (required)")
+		"bundle image reference to track - may be used multiple times (required when --git not provided)")
```
Comment (lcarva) [inline cmd/track/track_bundle.go:169]: Is this right? It's required if neither `--git` nor `--input` are provided. Maybe we should just remove the "required" text from this description.

File: internal/tracker/tracker.go [hunk 3]
```diff
@@ -154,11 +154,52 @@ func Track(ctx context.Context, urls []string, input []byte, prune bool, freshen
...
+			gits = append(gits, u)
+		} else {
+			imgs = append(imgs, u)
+		}
+	}
+
+	return imgs, gits
+}
+
+func trackImageReferences(ctx context.Context, urls []string, t *Tracker, freshen bool) error {
```
Comment (lcarva) [inline internal/tracker/tracker.go:192]: NIT: Why not make this a method on the `Tracker` type? (Same for `trackGitReferences`)

**Review by lcarva — APPROVED** [review]

File: internal/tracker/tracker.go (hunk 1 above)
Comment (zregvart) [reply internal/tracker/tracker.go:233]: Interesting, when handling on the policy side we might ignore whether the scheme is `git+ssh` or `git+https` or `git+...`, we might not care at all about the transfer protocol...

**Comment by zregvart** [comment]
Nits were easy to address so merging this now.

**Pull request merged**